dev/reloadable-server/src/main/scala/play/core/server/LagomReloadableDevServerStart.scala

```diff
@@ -47,43 +47,52 @@ object LagomReloadableDevServerStart {
         val process = new RealServerProcess(args = Seq.empty)
         val path: File = buildLink.projectPath
 
-        val keystoreBaseFolder = new File(".")
-        val keystoreFilePath = FakeKeyStore.getKeyStoreFilePath(keystoreBaseFolder)
-        FakeKeyStore.createKeyStore(keystoreBaseFolder)
+        val enableSsl = httpsPort > 0
 
         // The pairs play.server.httpx.{address,port} are read from PlayRegisterWithServiceRegistry
         // to register the service
         val httpsSettings: Map[String, String] =
-          Map(
-            // In dev mode, `play.server.https.address` and `play.server.http.address` are assigned the same value
-            // but both settings are set in case some code specifically read one config setting or the other.
-            "play.server.https.address" -> httpAddress, // there's no httpsAddress
-            "play.server.https.port" -> httpsPort.toString,
-            "play.server.https.keyStore.path" -> keystoreFilePath.getAbsolutePath,
-            "play.server.https.keyStore.type" -> "JKS",
-            "ssl-config.loose.disableHostnameVerification" -> "true"
-          )
+          if (enableSsl) {
+            val keystoreBaseFolder = new File(".")
+            val keystoreFilePath = FakeKeyStore.getKeyStoreFilePath(keystoreBaseFolder)
+            FakeKeyStore.createKeyStore(keystoreBaseFolder)
+
+            Map(
+              // In dev mode, `play.server.https.address` and `play.server.http.address` are assigned the same value
+              // but both settings are set in case some code specifically read one config setting or the other.
+              "play.server.https.address" -> httpAddress, // there's no httpsAddress
+              "play.server.https.port" -> httpsPort.toString,
+              "play.server.https.keyStore.path" -> keystoreFilePath.getAbsolutePath,
+              "play.server.https.keyStore.type" -> "JKS"
+            )
+          } else Map.empty
+
         val httpSettings: Map[String, String] =
           Map(
             // The pairs play.server.httpx.{address,port} are read from PlayRegisterWithServiceRegistry
             // to register the service
             "play.server.http.address" -> httpAddress,
-            "play.server.http.port" -> httpPort.toString
+            "play.server.http.port" -> httpPort.toString,
+            "ssl-config.loose.disableHostnameVerification" -> "true"
```

Comment:
shouldn't that be added only when ssl is enabled?

Reply:
If I remember correctly, this tunes the certificate validation on the client side.

If it's possible to enable SSL on a single service-A, any consuming end of service-A needs that setting (even if the consumer service doesn't bind SSL).

Reply:
If I understand it correctly, it means that we are enabling it everywhere because we don't know when a service will need it or not?

So, if ServiceA, which does not have ssl enable, needs to consume from ServiceB which has ssl enable, ServiceA needs that config. Is that correct?

If that's the case, it's maybe confusing to configure it altogether with the http setttings. It gives the impression that we are configuering the http server with this ssl-config.

Reply:
> If that's the case, it's maybe confusing to configure it altogether with the http setttings. It gives the impression that we are configuering the http server with this ssl-config.

Agreed. If my description above of what this setting does is accurate, we should extract the setting to `httpsClientSettings`.

I just realised that it will have an impact on users accessing remote, unmanaged services using HTTPS. We should add a disclaimer in the docs about this behavior being enabled by default.

Reply:
done: 47ff9bebe